File: solutions/252.meeting-rooms/meeting-rooms.js
/**
 * Definition for an interval.
 * function Interval(start, end) {
 *     this.start = start;
 *     this.end = end;
 * }
 */
/**
 * @param {Interval[]} intervals
 * @return {boolean}
 */
const canAttendMeetings = (intervals) => {
  let flag = true
  
  intervals.sort((a,b)=>a.start - b.start)
    
  intervals.reduce((acc,e)=>{
    if(e.start < acc) {
      flag = false
    }
    acc = e.end
    return acc
  }, 0)

  return flag
}

File: solutions/252.meeting-rooms/meeting-rooms.py
# Definition for an interval.
# class Interval:
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e


class Solution:
    def canAttendMeetings(self, intervals):
        """
        :type intervals: List[Interval]
        :rtype: bool
        """
        intervals.sort(key=lambda x: x.start)

        cur = float('-inf')
        for interval in intervals:
            if interval.start < cur:
                return False

            cur = interval.end

        return True
